src/engine/uiManager.hxx

```diff
@@ -53,6 +59,8 @@ private:
 
   std::unordered_map<std::string, std::vector<UiElement *>> m_uiGroups;
   std::unordered_map<std::string, ButtonGroup> m_buttonGroups;
+
+  friend class EventManager;
```

Comment:
Is not there a way not to use `friend`?

Reply:
I thought it's better to make EventManager a friend than to expost those 3 internal containers just for the event manager. Do you think i should rather make read-only getters for those?

Reply:
I've addressed this in the next PR with 7dfb8ce because there were a few changes to those internal containers in the next PR. 
Please select this as resolved and merge the PR, if it's ok with you

Reply:
Alright. I see.